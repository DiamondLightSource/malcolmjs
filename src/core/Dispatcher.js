/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
