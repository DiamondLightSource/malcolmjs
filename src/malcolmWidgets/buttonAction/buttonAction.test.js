import React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import ButtonAction from './buttonAction.component';

describe('ButtonAction', () => {
  let shallow;
  let mount;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <ButtonAction text="View" clickAction={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls clickAction when clicked', () => {
    const clickAction = jest.fn();
    const wrapper = mount(
      <ButtonAction text="View" clickAction={clickAction} />
    );

    wrapper.find('button').simulate('click');

    expect(clickAction.mock.calls.length).toEqual(1);
  });
});
