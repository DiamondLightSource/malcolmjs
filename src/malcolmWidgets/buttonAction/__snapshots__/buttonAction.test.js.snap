// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonAction renders correctly 1`] = `
<div
  className="ButtonAction-container-1"
>
  <WithStyles(Button)
    className="ButtonAction-button-2"
    color="primary"
    onClick={[Function]}
    variant="raised"
  >
    View
  </WithStyles(Button)>
</div>
`;
